hacksoc_org/filters.py

```diff
@@ -0,0 +1,246 @@
+    """
+    return render_markdown(caller)
+
+
+@app.template_filter()
+def from_yaml(caller: str) -> Union[List, Dict]:
+    """Parses the argument as YAML. Useful for quickly defining data to pass to Jinja macros to
+    template.
+
+    ```
+    {% set events | from_yaml %}
+        - name: Boardgames and Cake
+          desc: >
+            In one of the oldest and noblest traditions of HackSoc, we descend upon a room, play
+            many boardgames, and eat much cake. Both boardgames and cake are brought along by
+            members, so if you do enjoy the events, please consider bringing something along.
+        - name: CoffeeScript
+          desc: >
+            An unprecedented case of joviality and camaraderie has struck HackSoc, and we have
+            responded with a weekly social over in the Ron Cooke Hub cafe on Hes East. Come for
+            lunch, a chat, and maybe a small game or two.
+    {% endset %}
+
+    {% for item in events %}
+        <section class="event">
+            <h2> {{ item.name }} </h2>
+            <p> {{ item.desc }} </p>
+        </section>
+    {% endfor %}
+    ```
+
+    Args:
+        caller (str): YAML source, usually from a Jinja block (eg block assignment or macro call
+        block)
+
+    Returns:
+        Union[List, Dict]: YAML supports top-level lists or dictionaries (objects).
+    """
+    return yaml.safe_load(caller)
+
+
+@app.template_filter()
+def split_lede(caller) -> Dict[str, str]:
+    """Parses HTML with regular expressions.
+
+    Splits the first paragraph of a news article from the rest of the article. Assumes that caller
+    starts with a <p> block and no other (non-whitespace) characters.
+
+    Args:
+        caller (str): HTML source beginning with a <p> block
+
+    Returns:
+        Dict[str,str]:
+            `lede`: First <p> of `caller`, **excluding** the surrounding tags
+            `text`: Rest of the article (everything after the first </p>)
+    """
+    lede_re = re.compile(r"^\s*<p>(.*?)</p>", flags=re.DOTALL)
+    match = lede_re.match(caller)
+
+    if match is None:
+        print(caller)
+        return {"lede": "", "text": caller}
+    else:
+        return {"lede": match[1], "text": caller[match.end(0) :]}
+
+
+months = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+]
+
+
+@app.template_filter()
+def format_date(d: date):
+    """Formats dates as "January 02, 2021"
+
+    Args:
+        d (date): Python datetime.date object
+
+    Returns:
+        str: formatted date
+    """
+    if not isinstance(d, date):
+        # this can happen when calling get_template_attribute on an article
+        # since the date doesn't get passed to the template at that point, only in the
+        # render_article route.
+        return str(d)
+    else:
+        return f"{months[d.month-1]} {d.day:02d}, {d.year}"
```

Comment:
suggestion (nonblocking): `date.strftime("%B %d, %Y")`, which avoids the need for `months` completely

but that also depends on the current locale, so your approach is probably better tbh

Reply:
yeah, unfortunately there doesn't seem to be a good crossplatform way of setting the locale such that the months are in english.